server/protocols/jmap/src/main/java/org/apache/james/jmap/json/ObjectMapperFactory.java

```diff
@@ -102,6 +106,24 @@ public void serialize(MailboxId value, JsonGenerator gen, SerializerProvider ser
         }
     }
 
+    public static class UsernameKeyDeserializer extends KeyDeserializer {
+        @Override
+        public Object deserializeKey(String key, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+            return new Rights.Username(key);
+        }
+    }
+
+    public static class RightDeserializer extends JsonDeserializer<Rights.Right> {
+
+        @Override
+        public Rights.Right deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+            String nodeValue = p.getValueAsString();
+            Preconditions.checkArgument(nodeValue.length() == 1);
```

Comment:
what happens else? is there an integration test for that?

Reply:
idem when right is unknown

Reply:
I added unit tests on deserialization layer.

Also added integration tests. It gets rejected with a invalidArguments error type in the JMAP layer.